lens/sqlrepo/repo.go

```diff
@@ -1,17 +1,60 @@
 package sqlrepo
 
 import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
 	"github.com/urfave/cli/v2"
 
+	pgchainbs "github.com/filecoin-project/go-bs-postgres-chainnotated"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/lotus/chain/types"
 	"github.com/filecoin-project/sentinel-visor/lens"
 	"github.com/filecoin-project/sentinel-visor/lens/util"
 )
 
 func NewAPIOpener(c *cli.Context) (lens.APIOpener, lens.APICloser, error) {
-	bs, err := NewBlockStore(c.String("repo"))
+	pgbsCfg := pgchainbs.PgBlockstoreConfig{
+		PgxConnectString:        c.String("repo"),
+		CacheInactiveBeforeRead: true,
+		DisableBlocklinkParsing: true,
+		LogCacheStatsOnUSR1:     true,
+	}
+
+	pgbsCfg.InstanceNamespace = c.String("lens-postgres-namespace")
+	if pgbsCfg.InstanceNamespace == "" {
+		pgbsCfg.InstanceNamespace = "main"
+	}
+
+	if customCacheSize := c.Int("lens-cache-hint"); customCacheSize != 1024*1024 {
```

Comment:
@willscott what is the meaning of `1024*1024` in this context?

Reply:
it's the default value on the flag that's already being used by other lenses for an internal object cache